### PR TITLE
Assertions and Design By Contract in Java

### LINKED ISSUE #2659: Assertion and design by contract in java
### Proposed title of article
Assertion and design by contract in java
### Introduction paragraph (2-3 paragraphs):
The design by contract specifies the interaction between various components of an application. To achieve this it defines a contract based on which the components of the application communicate with each other.

The design by contract technique uses assertions to check whether the application meets the requirement specified in the defined contract. These assertions can be used to test the assumptions made by a programmer in java programming language.
### Key takeaways: 
1. Design by contract.
2. Implementing assertion.
3. Assertion rules.
4. How to create a java program using assertion.
### References: 
N/A

### PR DESCRIPTION
**Pull request by Paul22691**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ]Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2659

### CODE REVIEW

**Comment by Dawe-7** [comment]
Hello @Paul22691, I'll be reviewing your article. I noticed that you have long sentences that are hard to read. Before we begin the review, I'll need you to use the [hemimgway editor](https://hemingwayapp.com/rl) to check on that. Also, your hero image should be saved as `hero.jpg` and not `hero.jpg.jpg`

**Comment by Dawe-7** [comment]
@Paul22691, Kindly add more content in the conclusion and revisit the article once more.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by Paul22691** [comment]
@Dawe-7 you can now proceed

File: content/articles/assertion-and-design-by-contract-in-java/index.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+  - url: /engineering-education/assertion-and-design-by-contract-in-java/hero.jpg
+    alt: Assertion and design by contract in Java
+---
+The technique of design by contract specifies the interaction between various components of an application. This concept was first discovered and introduced in the Eiffel programming language. 
+ <!--more-->
+For interaction between various components to be achieved, a contract based on the components of the application to communicate with each other is defined by the concept. Assertions are used in the Design by contract technique to check if the requirements specified in the defined contract are met by the application. They are used to test the assumptions made by the programmer.
+
+### prerequisites
+Before reading this tutorial, you need to have basic knowledge of Java.
```
Comment (Dawe-7) [inline content/articles/assertion-and-design-by-contract-in-java/index.md:22]: Can you add an IDE link where readers can run the Java program

File: content/articles/assertion-and-design-by-contract-in-java/index.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+1. Checking the method arguments passed.
+2. Using the assertion in the default case of the switch statement.
+3. Making use of an assertion descriptive.
+4. Avoid processing in an assertion condition.
+5. Avoid catching an assertion-related exception.
+6. Avoid the use of more than one condition in an assertion
+
+### Creating a Java program
+This program helps the learner to implement assertion by dividing two numbers.
```
Comment (Dawe-7) [inline content/articles/assertion-and-design-by-contract-in-java/index.md:126]: Here, we need to add some comments in the program and an explanation of what is happening after the program

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
@Paul22691, Implement the suggestions so that we can move on to final review.

**Comment by Paul22691** [comment]
@Dawe-7 done on the changes lets proceed

**Review by Dawe-7 — APPROVED** [review]

**Comment by LinusMuema** [comment]
Hi @Paul22691 
Thanks for the article. Keep up the good work.

cc: @Dawe-7 
Sorry it took a while to make the final review

**Pull request merged**